infertrade/utilities/operations.py

```diff
@@ -37,30 +37,40 @@
 
 def pct_chg(x: Union[np.ndarray, pd.Series]) -> np.ndarray:
     """Percentage change between the current and a prior element."""
-    x = x.astype("float64")
-
-    if isinstance(x, pd.DataFrame):
-        pc = x.pct_change().values.reshape(-1, 1)
+    if not isinstance(x, (pd.DataFrame, pd.Series, np.ndarray)):
+        raise TypeError("must be Pandas Series, DataFrame or numpy ndarray")
```

Comment:
Arg hint is not consistent with the instance check.

Also - why do we want to support DataFrames and Series? Is that necessary? If a DataFrame is supplied we do not know which column we need to take changes on.

Reply:
> why do we want to support DataFrames and Series? Is that necessary? If a DataFrame is supplied we do not know which column we need to take changes on.

yes, that makes sense we should only support Pandas Series and Numpy ndarray. Note that support for Pandas DataFrame is copied from main branch so I didn't made any changes to that. But I will make this change, as it seems unnecessary to provide support for DataFrame.